src/serdes/binary-serializer.ts

```diff
@@ -133,11 +133,6 @@ class BinarySerializer {
       this.writeLengthEncoded(associatedValue);
     } else {
       associatedValue.toBytesSink(this.sink);
```

Comment:
Can you discuss why this logic is safe to remove?

Reply:
Yeah, this looks a bit off, moving the end markers into the internal bytes representation of the arrays 
I guess it works though :)

It's been many moons, but I don't think there were/are (at least *currently*) any instances of an STObject with a direct child STObject field.  

Maybe transaction metadata ? I wouldn't have 100% faith in the test coverage here.

Reply:
Would definitely check more transaction meta data examples

Reply:
Alternative would be to move the logic back into the BinarySerializer, and then use BinaarySerializer to compute the internal bytes representation of STArray and STObject